cmd/limactl/protect.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/lima-vm/lima/pkg/store"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+func newProtectCommand() *cobra.Command {
+	var protectCommand = &cobra.Command{
+		Use:   "protect INSTANCE [INSTANCE, ...]",
+		Short: "Protect an instance to prohibit accidental removal",
+		Long: `Protect an instance to prohibit accidental removal via the 'limactl delete' command.
+An instance can be still removable via '/bin/rm', Finder, etc.`,
+		Args:              WrapArgsError(cobra.MinimumNArgs(1)),
+		RunE:              protectAction,
+		ValidArgsFunction: protectBashComplete,
+	}
+	return protectCommand
+}
+
+func protectAction(_ *cobra.Command, args []string) error {
+	for _, instName := range args {
+		inst, err := store.Inspect(instName)
+		if err != nil {
+			return fmt.Errorf("failed to inspect instance %q: %w", instName, err)
+		}
+		if err := inst.Protect(); err != nil {
+			return fmt.Errorf("failed to protect instance %q: %w", instName, err)
+		}
```

Comment:
Why does processing stop on the first error? So `limactl protect foo missing bar` will protect `foo` but not `bar` because `missing` did not exist?

I think it would be better to collect all errors and report them at the end.

Reply:
Fixed